src/services/etherspot.js

```diff
@@ -96,19 +95,24 @@ export class EtherspotService {
      * Cycle through the supported networks and build an
      * array of instantiated instances
      */
-    this.supportedNetworks.forEach((networkName) => {
-      const env = networkName !== NetworkNames.Kovan ? EnvNames.MainNets : EnvNames.TestNets;
-      this.instances[networkName] = new EtherspotSdk(privateKey, { env, networkName });
+    await Promise.all(this.supportedNetworks.map(async (networkName) => {
+      try {
+        const env = networkName !== NetworkNames.Kovan ? EnvNames.MainNets : EnvNames.TestNets;
+        this.instances[networkName] = new EtherspotSdk(privateKey, { env, networkName });
+
+        // FCM only for mainnet, session creation should happen before computing contract account
+        if (fcmToken && networkName === primaryNetworkName) {
+          await this.instances[networkName].createSession({ fcmToken });
+        }
 
-      // Schedule exection of computeContractAccount's
-      etherspotComputeContractPromises.push(this.instances[networkName].computeContractAccount({ sync: true }));
-    });
+        await this.instances[networkName].computeContractAccount({ sync: true });
+      } catch (error) {
+        reportErrorLog('EtherspotService network init failed', { networkName, error });
+      }
+    }));
 
```

Comment:
I'm seeing this pattern emerge where we have nested `await`s, `try / catch`es, wrapped by another `await`s - i'm just a bit worried that this is starting to get messy, complex and less readable. I think whilst technically it works - we need to have a wider discussion about how we measure cyclomatic complexity within our code after V2 launch.

Reply:
I do agree, it needs wider discussion. However, separated 2 try catch's for each `await`.